src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -175,6 +181,23 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
       }
 
       case Types.NUMERIC:
+        if (numericPrecisionMapping) {
+          int precision = metadata.getPrecision(col);
+          if (metadata.getScale(col) == 0 && precision < 20) { // integer
```

Comment:
To take int8 as an example, for storing int8 `NUMBER(2,0)` does not work as you said since there are 3 digits of precision for `[-128..127]`, so if you were to create a db column for storing int8 `NUMBER(3,0)` makes sense.

But if you are getting a `NUMBER(3,0)` value back from the db, in theory the value could be anywhere in `[-999,999]`, so it would be unsafe to treat that as an int8.